### ISSUE TITLE
Application need reconnect

### ISSUE THREAD

**Issue by matav** [isoos/postgresql-dart#37]
I have **Dart** with **self_io** application on server. **Postgres in docker**. Once in a while the database has a problem and restarts. It recovers and closes the application connection after maybe 14 days of connecting to the database. 
The database recovers in seconds, but the Dart application can no longer connect and needs to be restored.

Dart error: 

`[ERROR] 2022-05-14 14:07:40.579992        0:00:31.096787        POST        /upload`
`PostgreSQLSeverity.error : Query cancelled due to the database connection closing.`
`package:postgres/src/connection.dart 453           _PostgreSQLExecutionContextMixin._query`
`package:postgres/src/connection.dart 427           _PostgreSQLExecutionContextMixin.query`

`
void main(List<String> args) async {
  final ip = InternetAddress.anyIPv4;
  final _handler = Pipeline().addMiddleware(logRequests()).addHandler(_router);

  connection = PostgreSQLConnection("localhost", 5432, "db1", username: "root", password: "password");
  await connection.open();

  final port = int.parse(Platform.environment['PORT'] ?? '4568');
  final server = await serve(_handler, ip, port);
  print('Server listening on port ${server.port}');
}
`

I need to set up the database connection so that it can automatically reconnect without resetting the application.

Best regards

**Comment by isoos** [comment]
@matav: I have a separate package for connection pool and lifecycle management:
https://pub.dev/packages/postgres_pool

There is a plan to eventually merge it with `package:postgres` and treat it as a transparent feature, but I'm not there yet....

**Comment by mhmzdev** [comment]
Any updates on this?
Facing same issue for any request made twice

**Comment by Pingear** [comment]
@isoos Yes, I saw that the pool is already included in postgres called experimental version 3, what is the status of the experiment, will there be documentation and when will you announce that everything works as it should with the pool out of the box, please tell us more,  we are very much waiting))
